### PR TITLE
Use a more recent kernel for Thinkpad X1

### PR DESCRIPTION
**Pull request by mrkkrp**
This solves lagging noticeable on high-resolution screens.

### CODE REVIEW

File: lenovo/thinkpad/x1/default.nix [hunk 1]
```diff
@@ -1,6 +1,9 @@
+{ lib, pkgs, ... }:
+
 {
   imports = [
     ../.
     ../../../common/cpu/intel
   ];
+  boot.kernelPackages = pkgs.linuxPackages_latest;
```
Comment (Mic92) [inline lenovo/thinkpad/x1/default.nix:8]: Let's make this future proof.

```suggestion
  # This solves lagging noticeable on high-resolution screens.
  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.15")  pkgs.linuxPackages_latest;
```

**Comment by Mic92** [comment]
Which generation of x1 are we talking about?

**Comment by mrkkrp** [comment]
The model I have is ThinkPad X1 Carbon Gen 9. Should we put this in a separate directory?

**Comment by mrkkrp** [comment]
@Mic92 Is there something else that should be done for this PR to become mergeable?

File: lenovo/thinkpad/x1/default.nix [hunk 2]
```diff
@@ -1,6 +1,10 @@
+{ lib, pkgs, ... }:
+
 {
   imports = [
     ../.
     ../../../common/cpu/intel
   ];
+  # This solves lagging noticeable on high-resolution screens.
+  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.15")  pkgs.linuxPackages_latest;
```
Comment (Mic92) [inline lenovo/thinkpad/x1/default.nix:9]:
```suggestion
  boot.kernelPackages = lib.mkIf (lib.versionOlder pkgs.linux.version "5.15")  (lib.mkDefault pkgs.linuxPackages_latest);
```

**Comment by Mic92** [comment]
> The model I have is ThinkPad X1 Carbon Gen 9. Should we put this in a separate directory?

That might be a good idea. X1 has been on the market for a while. We probably should not force long-term users of older devices to a new kernel since this might make it harder to use some out-of-tree kernel modules.

**Comment by mrkkrp** [comment]
Okay, I have squashed the commits and instead of modifying the shared config I created a `9th-gen` subdirectory.

File: lenovo/thinkpad/x1/9th-gen/default.nix [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{ lib, pkgs, ... }:
+{
+  imports = [
+    ../.
+  ];
```
Comment (Mic92) [inline lenovo/thinkpad/x1/9th-gen/default.nix:5]: Can you also reference this profile in the README as well as the flake?

Comment (mrkkrp) [reply lenovo/thinkpad/x1/9th-gen/default.nix:5]: Done.

**Pull request merged**